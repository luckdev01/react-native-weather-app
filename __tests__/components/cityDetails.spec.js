import 'react-native';
import React from 'react';
import {View, TextInput, Text, Button, TouchableHighlight} from 'react-native';
import { shallow } from 'enzyme';
import CityDetails from '../../src/components/cityDetails';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<CityDetails />', () => {
    it('Input and Button component renders correctly', () => {
      const tree = renderer.create(
        <CityDetails city={{weatherInfo: {}}}/>
      );
      const json = tree.toJSON();
      expect(json).toMatchSnapshot();
    });
    const wrapper = shallow(<CityDetails city={{weatherInfo: {}}}/>);

    it('Should exists', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('Should have two views', () => {
        expect(wrapper.find(View)).toHaveLength(2);
    });

    it('Should have one touchablehighlight', () => {
        expect(wrapper.find(TouchableHighlight)).toHaveLength(1);
    });

    it('Should have three texts', () => {
        expect(wrapper.find(Text)).toHaveLength(3);
    });
});
