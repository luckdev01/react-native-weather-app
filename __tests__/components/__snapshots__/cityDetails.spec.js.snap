exports[`<CityDetails /> Input and Button component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#262626",
      "flex": 1,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "marginLeft": 8,
          "marginTop": 30,
        },
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 20,
        }
      }>
      < Back
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 150,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Nunito-Bold",
          "fontSize": 50,
          "height": 70,
          "textAlign": "center",
        }
      }>
      undefined undefinedÂº
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Nunito-Bold",
          "fontSize": 40,
          "textAlign": "center",
        }
      } />
  </View>
</View>
`;
